File: app/src/main/java/org/stepic/droid/util/resolvers/SearchResolverImpl.java
package org.stepic.droid.util.resolvers;

import org.jetbrains.annotations.NotNull;
import org.stepic.droid.model.SearchResult;

import java.util.ArrayList;
import java.util.List;

public class SearchResolverImpl implements SearchResolver {

    @NotNull
    @Override
    public long[] getCourseIdsFromSearchResults(List<SearchResult> searchResultList) {
        List<Long> courseIds = new ArrayList<>();
        if (searchResultList == null || searchResultList.size() == 0) return new long[0];

        for (SearchResult item : searchResultList) {
            long courseId = item.getCourse();
            if (courseId != 0) {
                courseIds.add(courseId);
            }
        }

        long[] result = new long[courseIds.size()];
        for (int i = 0; i < result.length; i++) {
            result[i] = courseIds.get(i);
        }
        return result;
    }
}

File: app/src/main/java/org/stepic/droid/util/resolvers/SearchResolverImpl.kt
package org.stepic.droid.util.resolvers

import org.stepik.android.model.SearchResult
import javax.inject.Inject

class SearchResolverImpl @Inject constructor() : SearchResolver {
    override fun getCourseIdsFromSearchResults(searchResultList: List<SearchResult>?): LongArray =
        searchResultList
            ?.filter { it.course > 0L }
            ?.map { it.course }
            ?.toLongArray()
            ?: longArrayOf()
}
